napari/layers/utils/layer_utils.py

```
@@ -61,6 +63,68 @@ def _inner(func):
     return _inner
 
 
+def register_label_attr_action(
+    keymapprovider,
+    description: str,
+    attribute_name: str,
+    shortcuts=None,
+):
+    """
+    Convenient decorator to register an action with the current Layers
+
+    It will use the function name as the action name. We force the description
+    to be given instead of function docstring for translation purpose.
+
+    Parameters
+    ----------
+    keymapprovider : KeymapProvider
+        class on which to register the keybindings – this will typically be
+        the instance in focus that will handle the keyboard shortcut.
+    description : str
+        The description of the action, this will typically be translated and
+        will be what will be used in tooltips.
+    attribute_name : str
+        The name of the attribute to be restored if key is hold over `get_settings().get_settings().application.hold_button_delay.
+    shortcuts : str | List[str]
+        Shortcut to bind by default to the action we are registering.
+
+    Returns
+    -------
+    function:
+        Actual decorator to apply to a function. Given decorator returns the
+        function unmodified to allow decorator stacking.
+
+    """
+
+    def _handle(func):
+        sig = inspect.signature(func)
+        try:
+            name = next(iter(sig.parameters))
+        except StopIteration:
+            return register_layer_action(
```

@Czaki what is this meant to cover? (Since codecov complains that tests don't cover it, it's unclear.)

"name" is also unclear as a variable name. I see below that this is probably `first_parameter_name`? Why is only the first parameter name important here? I think some assumptions of the function should be spelled out in the docstring.

---

update to exception instead of silent fail.